Sources/Clappr_iOS/Classes/Plugin/Core/Drawer/BottomDrawerPlugin.swift

```diff
@@ -167,32 +145,35 @@ open class BottomDrawerPlugin: DrawerPlugin {
         guard let view = gesture.view else { return }
 
         let translation = gesture.translation(in: view)
-        let newBottomDistance = topDistanceFromBottom.constant + translation.y
+        let newBottomDistance = drawerTopConstraint.constant + translation.y
 
         if canDrag(with: newBottomDistance) {
-            topDistanceFromBottom.constant = newBottomDistance
+            drawerTopConstraint.constant = newBottomDistance
 
-            let maxOpacity: CGFloat = 1.0
-            let distanceFromBottom = abs(newBottomDistance)
-            let portionShown = distanceFromBottom / height
-
-            let mediaControlAlpha = maxOpacity - portionShown
-            core?.trigger(InternalEvent.didDragDrawer.rawValue, userInfo: ["alpha": mediaControlAlpha])
+            calculateMediaControlAlpha(for: newBottomDistance)
         }
     }
 
+    private func calculateMediaControlAlpha(for newBottomDistance: CGFloat) {
+        let maxOpacity: CGFloat = 1.0
+        let distanceFromBottom = abs(newBottomDistance)
+        let portionShown = distanceFromBottom / actualHeight
+        let mediaControlAlpha = maxOpacity - portionShown
+
+        core?.trigger(InternalEvent.didDragDrawer.rawValue, userInfo: ["alpha": mediaControlAlpha])
```

Comment:
I suggest we pull this line out of this method, returning the desired alpha instead. As it is, this method is handling two responsibilities. Also sending an event in a method `calculateMediaControlAlpha:for:` is a really unexpected side effect.